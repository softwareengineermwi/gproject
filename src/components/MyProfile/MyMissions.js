import React from 'react';
import { useSelector } from 'react-redux';

const MyMissions = () => {
  const myMissions = useSelector((state) => state.missions.filter((mission) => mission.joined));
  return (
    <div className="col-6">
      <h3>My Missions</h3>
      <table className="border width-100%">
        <tbody>
          {
            myMissions.map((mission) => (
              <tr key={mission.id} className="bg-white">
                <td>{mission.name}</td>
              </tr>
            ))
          }
        </tbody>
      </table>
    </div>
  );
};

export default MyMissions;
