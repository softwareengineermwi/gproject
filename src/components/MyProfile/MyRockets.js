import React from 'react';
import { useSelector } from 'react-redux';

const filtrate = (item) => item.reserved === true;
const MyRockets = () => {
  const t = useSelector((state) => state.rockets).filter(filtrate);
  return (
    <div className="col-6">
      <h3>My Rockets</h3>

      <table className="border width-100%">
        <tbody>
          {
            t.map((rocket) => (
              <tr className="padding-lg" key={rocket.id}>
                <td>{rocket.name}</td>
              </tr>
            ))
          }
        </tbody>
      </table>
    </div>
  );
};

export default MyRockets;
