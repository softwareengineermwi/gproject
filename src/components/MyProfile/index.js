import React from 'react';
import MyMissions from './MyMissions';
import MyRockets from './MyRockets';

const MyProfile = function () {
  return (
    <div className="container">
      <div className="grid gap-lg">
        <MyMissions />
        <MyRockets />
      </div>
    </div>
  );
};

export default MyProfile;
