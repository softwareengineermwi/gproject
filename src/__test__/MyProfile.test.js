import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../components/MyProfile';

describe('MyProfile Component test', () => {
  it('Should render correctly', () => {
    const app = renderer.create(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    ).toJSON();
    expect(app).toMatchSnapshot();
  });
});