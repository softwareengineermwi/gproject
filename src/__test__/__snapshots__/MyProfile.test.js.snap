// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyProfile Component test Should render correctly 1`] = `
<div
  className="container"
>
  <div
    className="grid gap-lg"
  >
    <div
      className="col-6"
    >
      <h3>
        My Missions
      </h3>
      <table
        className="border width-100%"
      >
        <tbody />
      </table>
    </div>
    <div
      className="col-6"
    >
      <h3>
        My Rockets
      </h3>
      <table
        className="border width-100%"
      >
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
